bufferserver/src/main/java/com/datatorrent/bufferserver/internal/LogicalNode.java

```diff
@@ -138,9 +140,7 @@ public boolean isReady()
     if (!ready) {
       ready = true;
       for (PhysicalNode pn : physicalNodes) {
-        if (pn.isBlocked()) {
-          ready = pn.unblock() & ready;
-        }
+        ready = pn.unblock() & ready;
```

Comment:
There should be only one check for `blocker == null`. PhysicalNode.unblock() encapsulates blocking state handling. There is no need to call PhysicalNode.isBlocked() here.